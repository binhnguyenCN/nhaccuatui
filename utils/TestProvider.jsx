// libs
import { useState } from 'react';
// others
import locales from '@/locales';
import { ProvideLocale } from '@/context/LocaleContext';

const TestProvider = ({ children }) => {
  const [locale, setLocale] = useState('vi');
  return (
    <ProvideLocale
      value={{ locale, setLocale, localDataSource: locales[locale] }}
    >
      {children}
    </ProvideLocale>
  );
};
export default TestProvider;
