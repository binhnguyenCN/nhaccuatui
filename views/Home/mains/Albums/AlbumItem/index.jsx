// libs
import Link from 'next/link';
import React from 'react';
// others
import styles from './styles.module.scss';

const AlbumItem = ({ title, thumbnail }) => (
  <li className={styles['album-item']}>
    <Link href={thumbnail} passHref>
      <span>
        <img
          src={thumbnail}
          width={152}
          height={152}
          alt="topic"
          title={title}
        />
      </span>
    </Link>
    <Link href={thumbnail} passHref>
      <span className={styles['album-title']} title={title}>
        {title.substring(0, 20)}
        {title.length > 40 && '...'}
      </span>
    </Link>
  </li>
);

export default AlbumItem;
