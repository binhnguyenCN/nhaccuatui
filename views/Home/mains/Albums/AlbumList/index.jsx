// libs
import React from 'react';
// layouts
// import styles from './styles.module.scss';

const AlbumList = () => <li>Item</li>;

export default AlbumList;
