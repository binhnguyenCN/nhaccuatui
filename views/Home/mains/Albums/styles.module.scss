.album-wrapper {
  &:first-child {
    grid-area: album1;
  }

  &:nth-child(2) {
    grid-area: album2;
  }

  &:nth-child(3) {
    grid-area: album3;
  }
}
