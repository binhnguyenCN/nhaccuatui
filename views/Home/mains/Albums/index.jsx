// libs
import React from 'react';
// layouts
import CategoryList from '../../components/CategoryList';
import CategoryTitle from '../../components/CategoryTitle';
import AlbumItem from './AlbumItem';
import styles from './styles.module.scss';

const Albums = ({ albums }) => {
  const { groupName, listPlaylist } = albums;
  const getTitle = (title, lang = 'vi') => {
    const index = title.indexOf('_');
    if (lang === 'vi') {
      return title.slice(0, index);
    }
    return title.slice(index + 1, -1);
  };
  return (
    <div className={styles['albums-wrapper']}>
      <CategoryTitle
        title={{ content: getTitle(groupName), url: '/', icon: null }}
      />
      <CategoryList>
        {listPlaylist.map((playlist, index) => {
          if (index < 5) {
            return <AlbumItem key={playlist.key} album={playlist} />;
          }
          return null;
        })}
      </CategoryList>
    </div>
  );
};

export default Albums;
