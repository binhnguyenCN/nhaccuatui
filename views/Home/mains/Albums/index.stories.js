// libs
import React from 'react';
// layouts
import Albums from '.';
// others
import mockData from '../../../../api/mockApi';

export default {
  title: 'Pages/Home/main/Albums',
  component: Albums,
  parameters: { albums: mockData.albumList[0] },
};
export const PrimaryAlbums = () => <Albums albums={mockData.albumList[0]} />;
