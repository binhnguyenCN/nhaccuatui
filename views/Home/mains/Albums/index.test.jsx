// libs
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
// layouts
import Albums from '.';
// mocks
import mockData from '@/mocks/mockApi';
// others
import TestProvider from '@/utils/TestProvider.jsx';

describe('render album list', () => {
  it('should include 5 items', () => {
    // arrange
    const testId = 'album-item';
    const maxLength = 5;
    // act
    render(
      <TestProvider>
        <Albums albums={mockData.albumList[0]} />,
      </TestProvider>,
    );
    // assert
    expect(screen.getAllByTestId(testId)).toHaveLength(maxLength);
  });
  it('should renders correctly', () => {
    // act
    const tree = renderer
      .create(
        <TestProvider>
          <Albums albums={mockData.albumList[0]} />,
        </TestProvider>,
      )
      .toJSON();
    // assert
    expect(tree).toMatchSnapshot();
  });
});
