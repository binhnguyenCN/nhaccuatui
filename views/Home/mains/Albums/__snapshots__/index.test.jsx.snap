// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render album list should renders correctly 1`] = `
Array [
  <div
    className="albums-wrapper"
  >
    <div
      className="albums-inner"
    >
      <h3
        className="category-title"
      >
        <span
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span>
            Điểm Sáng Nhạc Việt
          </span>
        </span>
      </h3>
      <ul
        className="category-list"
      >
        <li
          className="album-item"
          data-testid="album-item"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650365717216_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="topic"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650365717216_300.jpg"
              title="V-POP Những Bản Hit Quốc Dân"
              width={152}
            />
          </span>
          <span
            className="album-title"
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650365717216_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
            title="V-POP Những Bản Hit Quốc Dân"
          >
            V-POP Những Bản Hit 
          </span>
        </li>
        <li
          className="album-item"
          data-testid="album-item"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650365624994_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="topic"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650365624994_300.jpg"
              title="V-POP 2022: Hot Comeback"
              width={152}
            />
          </span>
          <span
            className="album-title"
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650365624994_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
            title="V-POP 2022: Hot Comeback"
          >
            V-POP 2022: Hot Come
          </span>
        </li>
        <li
          className="album-item"
          data-testid="album-item"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650339230854_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="topic"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650339230854_300.jpg"
              title="V-POP Today's Top Hits"
              width={152}
            />
          </span>
          <span
            className="album-title"
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/19/9/7/a/5/1650339230854_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
            title="V-POP Today's Top Hits"
          >
            V-POP Today's Top Hi
          </span>
        </li>
        <li
          className="album-item"
          data-testid="album-item"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/01/26/a/0/5/0/1643183067353_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="topic"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2022/01/26/a/0/5/0/1643183067353_300.jpg"
              title="Nhạc Rap Việt Hôm Nay Nghe Gì?"
              width={152}
            />
          </span>
          <span
            className="album-title"
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/01/26/a/0/5/0/1643183067353_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
            title="Nhạc Rap Việt Hôm Nay Nghe Gì?"
          >
            Nhạc Rap Việt Hôm Na
          </span>
        </li>
        <li
          className="album-item"
          data-testid="album-item"
        >
          <span
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/15/5/1/b/e/1650010670819_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <img
              alt="topic"
              height={152}
              src="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/15/5/1/b/e/1650010670819_300.jpg"
              title="Dễ Thương Quá Đi"
              width={152}
            />
          </span>
          <span
            className="album-title"
            href="https://avatar-ex-swe.nixcdn.com/playlist/2022/04/15/5/1/b/e/1650010670819_300.jpg"
            onClick={[Function]}
            onMouseEnter={[Function]}
            title="Dễ Thương Quá Đi"
          >
            Dễ Thương Quá Đi
          </span>
        </li>
      </ul>
    </div>
  </div>,
  ",",
]
`;
